src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/PgQuestionAttempts.java

```diff
@@ -395,24 +395,27 @@ public void mergeAnonymousQuestionInformationWithRegisteredUserRecord(final Stri
     }
 
     @Override
-    public Map<Role, Long> getAnsweredQuestionRolesOverPrevious(final TimeInterval timeInterval) throws SegueDatabaseException {
-        String query = "SELECT role, count(DISTINCT users.id) FROM question_attempts"
-                + " JOIN users ON user_id=users.id AND NOT deleted WHERE timestamp > now() - ? GROUP BY role";
-        try (Connection conn = database.getDatabaseConnection();
-             PreparedStatement pst = conn.prepareStatement(query)
-        ) {
-            pst.setObject(FIELD_GET_ANSWERED_QUESTION_ROLES_INTERVAL_AGO, timeInterval.getPGInterval());
-
-            try (ResultSet results = pst.executeQuery()) {
-                Map<Role, Long> resultsToReturn = Maps.newHashMap();
-                while (results.next()) {
-                    resultsToReturn.put(Role.valueOf(results.getString("role")), results.getLong("count"));
+    public Map<TimeInterval, Map<Role, Long>> getAnsweredQuestionRolesOverPrevious(final TimeInterval[] timeRanges) throws SegueDatabaseException {
+        Map<TimeInterval, Map<Role, Long>> allResults = new HashMap<>();
+        try (Connection conn = database.getDatabaseConnection()) {
+            for (TimeInterval timeRange : timeRanges) {
+                String query = "SELECT role, count(DISTINCT users.id) FROM question_attempts"
+                        + " JOIN users ON user_id=users.id AND NOT deleted WHERE timestamp > now() - ? GROUP BY role";
+                try (PreparedStatement pst = conn.prepareStatement(query)) {
+                    pst.setObject(1, timeRange.getPGInterval());
+                    try (ResultSet results = pst.executeQuery()) {
+                        Map<Role, Long> resultForTimeRange = new HashMap<>();
+                        while (results.next()) {
+                            resultForTimeRange.put(Role.valueOf(results.getString("role")), results.getLong("count"));
+                        }
+                        allResults.put(timeRange, resultForTimeRange);
+                    }
                 }
-                return resultsToReturn;
             }
-        } catch (SQLException e) {
+        }  catch (SQLException e) {
```

Comment:
Seem to have picked up an extra space here? You'll probably need to do a formatting run anyway when you merge in the new ruleset so may as well fix it then...